### ISSUE TITLE
Incorrect recurring total for free products with WC Subscriptions

### ISSUE THREAD

**Issue by malithsen** [Automattic/woocommerce-payments#4768]
### Describe the bug
Recurring total is inconsistent when changing the shipping method for a free subscription (not a free trial sub, just a subscription with 0$ fee)

### To Reproduce

1. Install and enable WooCommerce Subscriptions.
2. Create a new simple subscription product.
3. Make the subscription price $0.
4. Add two shipping methods, Flat rate and Free Shipping, via WooCommerce -> Settings -> Shipping
5. Make sure Flat rate shipping has a non-zero value.
6. As a customer add the free subscription to the cart and go to checkout.
7. Toggle between the two shipping methods.
8. Notice that the Recurring Total field shows non-zero sum when free shipping is selected
9. Refresh the page while Free Shipping is selected.
10. Notice the Recurring Total now shows $0

### Actual behavior

Inconsistent amount on Recurring Total field

### Screenshots

https://user-images.githubusercontent.com/6216000/190594920-e4abfb39-6486-42f2-a026-196f9679d81f.mov

### Expected behavior

Recurring total should be 0 for $0 subscriptions when using free shipping

**Comment by jessy-p** [comment]
Before starting check if related to or affected by https://github.com/Automattic/woocommerce-payments/issues/4764.

**Comment by perezcarreno** [comment]
Thank you, @malithsen, for reporting this. We see the same issue occur even with a non-zero subscription price.

**Comment by andymoreno** [comment]
Hello, any news on this issue? Thank you.

**Comment by zmaglica** [comment]
Moved back to the prioritization. Context: p1693316652722699-slack-C01BPL3ALGP

**Comment by jessy-p** [comment]
To be tested by the porter

**Comment by htdat** [comment]
> Make the subscription price $0.

It's not possible to add a subscription price is $0 though I could set a free trial for the first month. Even with a free trial, I could not replicate this issue with WCPay Subscriptions (with WooPayments 6.6.2), however, it might be an issue with WC Subscriptions: 

![Markup on 2023-10-31 at 11:28:20](https://github.com/Automattic/woocommerce-payments/assets/10045087/89526d99-0d30-43b5-bc22-d3ce762520a4)

I am asking team @Automattic/quark to have a further look (ref: p1698726398303719-slack-C055WHLA98D)

**Comment by james-allan** [comment]
Thanks for the ping @dat. 

This looks like a duplicate of https://github.com/woocommerce/woocommerce-subscriptions/issues/4335 which we fixed in `subscriptions-core` 5.9.0 (PR https://github.com/Automattic/woocommerce-subscriptions-core/pull/467) which was deployed in Woo Subscriptions and WooPayments in their July 2023 releases.

Given this issue was filed back in Sept last year and the most recent update was in February, I think we're good to close this as I cannot replicate it on WooPayments or Woo Subscriptions today.
